wavelength-plugins/roomshop.js

```diff
@@ -0,0 +1,211 @@
+
+		bank: {
+			set: function (target, room, user) {
+				target = toId(target);
+				if (!target && !target.registered) return this.errorReply(`The bank must be a registered user.`);
+				let roomshop = Db.roomshop.get(room.id, {items: {}});
+				if (!roomshop) return this.errorReply(`${room.title} does not have a roomshop yet.`);
+				if (roomshop.bank) return this.errorReply(`${room.title} already has a bank.`);
+				roomshop.bank = target;
+				Db.roomshop.set(room.id, roomshop);
+				this.privateModAction(`${room.title}'s Room Shop Bank has been set as "${target}" by ${user.name}.`);
+			},
+
+			remove: function (target, room, user) {
+				let roomshop = Db.roomshop.get(room.id, {items: {}});
+				if (!roomshop) return this.errorReply(`${room.title} does not have a roomshop yet.`);
+				if (!roomshop.bank) return this.errorReply(`${room.title} has no bank set yet.`);
+				delete roomshop.bank;
+				Db.roomshop.set(room.id, roomshop);
+				this.privateModAction(`${room.title}'s Room Shop Bank has been removed by ${user.name}.`);
+			},
+
+			account: "holder",
+			holder: function (target, room, user) {
+				if (!this.runBroadcast()) return;
+				let roomshop = Db.roomshop.get(room.id, {items: {}});
+				if (!Db.roomshop.has(room.id)) return this.errorReply(`${room.title} does not have a roomshop.`);
+				if (!roomshop.bank) return this.errorReply(`${room.title} hasn't set a bank yet.`);
+				return this.sendReplyBox(`<strong>${room.title}'s Bank:</strong> ${WL.nameColor(roomshop.bank, true)}`);
+			},
+
+			"": "atm",
+			money: "atm",
+			atm: function (target, room, user) {
+				if (!this.runBroadcast()) return;
+				let roomshop = Db.roomshop.get(room.id, {items: {}});
+				if (!roomshop.bank) return this.errorReply(`${room.title} hasn't set a bank yet.`);
+				if (this.broadcasting) {
+					this.parse(`!atm ${roomshop.bank}`);
+				} else {
+					this.parse(`/atm ${roomshop.bank}`);
+				}
+			},
+		},
+
+		additem: "add",
+		add: function (target, room, user) {
+			if (!this.can("roommod", null, room)) return false;
+			target = target.split(",").map(p => p.trim());
+			let roomshop = Db.roomshop.get(room.id, {items: {}});
+			if (Object.keys(roomshop.items).length > 12) return this.errorReply(`${room.title} has already reached the maximum amount of items allowed.`);
+			let item = target[0];
+			let price = target[1];
+			let desc = target[2];
+			if (!desc) return this.parse(`/help roomshop`);
+			if (item.length < 1 || item.length > 20) return this.errorReply(`The item name should be between 1-20 characters long.`);
+			if (isNaN(price)) return this.errorReply(`The price for the item must be an integer.`);
+			if (desc.length < 1 || desc.length > 50) return this.errorReply(`The description for an item must be between 1-50 characters long.`);
+			roomshop.items[toId(item)] = {id: toId(item), name: item, price: price, desc: desc};
```

Comment:
Don't forget that, with the other changes, the `desc` value needs to be `desc.join(', ')`.